app/controllers/videos_controller.rb

```diff
@@ -19,4 +19,20 @@ def toggle_star
     @video.toggle :starred
     @video.save
   end
+
+  #handle file uploads
+  def import   
+    if request.post?
+      file_extension = File.extname(params[:file].original_filename)
+        if file_extension == ".csv"
+          if Video.import(params[:file]) != false
+            redirect_to videos_path, notice: "Video(s) imported!"
```

Comment:
Do not add flash messages like this. I've added a flash message helper that allows multiple flash messages to be added. Use `add_flash(:alert, 'Videos imported!')`.